contracts/core/LyraVault.sol

```diff
@@ -58,29 +48,22 @@ contract LyraVault is Ownable, BaseVault {
       collateralAsset.approve(address(strategy), 0);
     }
 
-    strategy = DeltaStrategy(_strategy);
+    strategy = IStrategy(_strategy);
     collateralAsset.approve(_strategy, type(uint).max);
     emit StrategyUpdated(_strategy);
   }
 
   /// @dev anyone can trigger a trade
   function trade(uint strikeId) external {
     require(vaultState.roundInProgress, "round closed");
-    uint collateralBefore = collateralAsset.balanceOf(address(this));
-
     // perform trade through strategy
     (uint positionId, uint premiumReceived, uint collateralAdded) = strategy.doTrade(strikeId, lyraRewardRecipient);
-    uint collateralAfter = collateralAsset.balanceOf(address(this));
-
-    // verify used amount
-    uint assetUsed = collateralBefore - collateralAfter;
-    require(assetUsed <= collateralAdded, "improper collateral amount used");
```

Comment:
we don't need this check now because the amount is returned by the strategy, and it's designed by us to know the exact amount traded, and we can assume the strategy is honest.
(Previously it was here because the trade function won't return anything, and we want to make sure the amount is the same as the strategy "suggested" before the real trade happened)